DigitalLearningSolutions.Web/Controllers/ResetPasswordController.cs

```diff
@@ -0,0 +1,46 @@
+namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Services;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class ResetPasswordController : Controller
+    {
+        private readonly IPasswordResetService passwordResetService;
+
+        public ResetPasswordController(IPasswordResetService passwordResetService)
+        {
+            this.passwordResetService = passwordResetService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Index(string email, string code)
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            if (string.IsNullOrEmpty(email) || string.IsNullOrEmpty(code))
+            {
+                return RedirectToAction("Index", "Login");
+            }
+
+            var matchingUserRefs = await passwordResetService.GetValidMatchingUserReferencesAsync(email, code);

```

Comment:
I wonder if this method should return a boolean and be something like ResetPasswordLinkIsValid(email, code) - if we need the user later down the line, maybe we can shuffle things around within the passwordResetService. What do you think? :)

Reply:
In fact, when we reset a password (second part of the ticket), we don't care about whether the hash is correlated to the account entry or not - we should just change the pw for all the entries where the email matches the user email address as long as at least one of these entries matches the reset password hash (which is there to ensure that the user owns that email address).

Reply:
Yeah, so we will need the user id and type to be tracked in TempData or something for actually resetting the password. This is the result of the shuffling things around that I did because I started thinking about that part of the ticket last week.